### DISCUSSION TITLE
Azure AD and email authentication

### DISCUSSION THREAD

**Discussion by savornicesei** [WeblateOrg/weblate#6444] [Q&A]
Hi,
I've set up AAD and email login in Weblate. 
Kind of makes sense to require registration (REGISTRATION_OPEN=True) when an AAD user logs in for the first time (so a new user is added to Weblate) but that means registration is also open for email/password.

Is it possible to turn off just email/passwd registration in std. Weblate or this should be a customization?

Thanks.

All the best,
Simo

For those looking on setting up AAD login,  I've configured the following settings:
- SOCIAL_AUTH_AZUREAD_TENANT_OAUT2_KEY = <app id>
- SOCIAL_AUTH_AZUREAD_TENANT_OAUT2_KEY = <app key>
- SOCIAL_AUTH_AZUREAD_TENANT_OAUT2_KEY = <tenant id>
-  SOCIAL_AUTH_ADMIN_USER_SEARCH_FIELDS = ['username', 'first_name', 'last_name', 'email']
- SOCIAL_AUTH_POSTGRES_JSONFIELD = True
- SOCIAL_AUTH_REDIRECT_IS_HTTPS = True
while in AAD the return url was set to https://<weblate instance>/accounts/complete/azuread-tenant-oauth2/ Notice the ending slash. Https is a must for return url or AAD will throw an error.

**Comment by nijel** [comment]
You can remove the email/password authentication completely from `AUTHENTICATION_BACKENDS`.

You can also restrict which authentications can create new users by https://docs.weblate.org/en/latest/admin/config.html#std-setting-REGISTRATION_ALLOW_BACKENDS

**Reply by savornicesei** [reply]
Thank you. It works as it should now.
